airtai/faststream
# 1129 - Create a publish command for the CLI

# Description

Users should be able to publish a test message right from the command line. To do this, a `publish` command has been added to the CLI `main.py` file. 

Close  #1129

## How to Test

To test the new publish functionality introduced in this Pull Request, follow these steps:

1. **Start Your FastStream Application**: First, you need to have a running FastStream application. You can start your application with the following command, assuming your FastStream application is defined in a file named `testfaststreamapp.py`:

    ```bash
    python testfaststreamapp.py
    ```

    Make sure your FastStream application is set up to subscribe to a channel named `input_channel` and publishes processed messages to another channel (e.g., `output_channel`). Here's a simple example of what your `testfaststreamapp.py` might contain:

    ```python
    from faststream import FastStream
    from faststream.redis import RedisBroker

    broker = RedisBroker("redis://localhost:6379")
    app = FastStream(broker=broker)

    @broker.subscriber(channel="input_channel")
    @broker.publisher(channel="output_channel")
    async def process_message(message: str) -> str:
        print(f"Received message: {message}")
        return f"Processed: {message}"

    if __name__ == "__main__":
        app.run()
    ```

2. **Send a Test Message**: With your FastStream application running, you can send a test message using the `faststream publish` command. Use the following command to publish a message to the `input_channel`:

    ```bash
    faststream publish testfaststreamapp:app "Hello World" --channel "input_channel" --rpc
    ```

## Type of change


## Checklist

- [x] My code adheres to the style guidelines of this project (`scripts/lint.sh` shows no errors)
- [x] I have conducted a self-review of my own code
- [ ] I have made the necessary changes to the documentation
- [x] I have added tests to validate the effectiveness of my fix or the functionality of my new feature
- [ ] Both new and existing unit tests pass successfully on my local environment by running `scripts/test-cov.sh`
- [ ] I have ensured that static analysis tests are passing by running `scripts/static-anaylysis.sh`
- [ ] I have included code examples to illustrate the modifications
